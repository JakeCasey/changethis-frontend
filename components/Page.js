import React, { Component } from "react";
import styled, { ThemeProvider, injectGlobal } from "styled-components";
import Header from "./Header";
import Meta from "./Meta";

const theme = {
  primary: "lightskyblue",
  red: "#FF0000",
  black: "#393939",
  grey: "#3A3A3A",
  lightgrey: "#E1E1E1",
  offWhite: "#EDEDED",
  maxWidth: "100vw",
  bs: "0 5px 15px hsla(0, 0%, 0%, 0.2)",
  borders: "4px",
  button_borders: "4px",
  button_color: "lightskyblue"
};

const StyledPage = styled.div`
  background: white;
  color: ${props => props.theme.black};
`;

const Inner = styled.div`
  max-width: ${props => props.theme.maxWidth};
  margin: 0 auto;
  padding: 2rem;
`;

injectGlobal`
  /* @font-face {
    font-family: 'radnika_next';
    src: url('/static/radnikanext-medium-webfont.woff2') format('woff2');
    font-weight: normal;
    font-style: normal;
  } */

  @font-face {
    font-family: 'Inter';
    src: url('/static/Inter-Regular.woff') format('woff');
  }
  html {
    box-sizing: border-box;
    font-size: 10px;
  }
  *, *:before, *:after {
    box-sizing: inherit;
  }
  body {
    padding: 0;
    margin: 0;
    font-family: 'Inter', -apple-system, Segoe UI, Roboto, Noto Sans, Ubuntu, Cantarell, Helvetica Neue;
    /* font-family:  */
   /* font-family: 'radnika_next' */
  }
  a {
    text-decoration: none;
    color: ${theme.black};
  }

`;

class Page extends Component {
  render() {
    return (
      <ThemeProvider theme={theme}>
        <StyledPage>
          <Meta />
          <div className="p-12">
            <Header />
            {this.props.children}
          </div>
        </StyledPage>
      </ThemeProvider>
    );
  }
}

export default Page;
