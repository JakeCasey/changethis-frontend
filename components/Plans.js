import Error from '../components/ErrorMessage';
import GetPlansList from './wrappers/GetPlansList';
import SubscribeOrUnsubscribe from './SubscribeOrUnsubscribe';

const Plans = props => {
    return (
        <div>
            <GetPlansList>
                {({ data: { getPlansList }, error }) => {
                    if (error) return <Error error={error} />
                    return (
                        <div>
                            <h1> Our Plans </h1>
                            {getPlansList.map((p, i) => (<SubscribeOrUnsubscribe key={i} plan={p} />))}
                        </div>
                    )
                }}
            </GetPlansList>
        </div>
    );
};

export default Plans;
