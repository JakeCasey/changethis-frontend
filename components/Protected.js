import Error from '../components/ErrorMessage';
import gql from 'graphql-tag';

const Protected = props => {
  return (
    <div>
      <h1> This was protected </h1>
    </div>
  );
};

export default Protected;
