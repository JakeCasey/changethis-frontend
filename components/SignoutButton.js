import React, { Component } from "react";
import { Mutation } from "react-apollo";
import Error from "./ErrorMessage";
import gql from "graphql-tag";
import { CURRENT_USER_QUERY } from "./wrappers/User";

const SIGN_OUT_MUTATION = gql`
  mutation SIGN_OUT_MUTATION {
    signout {
      message
    }
  }
`;

export default class SignoutButton extends Component {
  render() {
    return (
      <Mutation
        mutation={SIGN_OUT_MUTATION}
        refetchQueries={[{ query: CURRENT_USER_QUERY }]}
      >
        {(signout, { loading, error }) => {
          return (
            <button onClick={() => signout()}>
              Sign Out{" "}
              <svg
                className="w-8 h-8 ml-4"
                xmlns="http://www.w3.org/2000/svg"
                viewBox="0 0 24 24"
              >
                <path
                  className="text-gray-600 fill-current"
                  d="M11 4h3a1 1 0 011 1v3a1 1 0 01-2 0V6h-2v12h2v-2a1 1 0 012 0v3a1 1 0 01-1 1h-3v1a1 1 0 01-1.27.96l-6.98-2A1 1 0 012 19V5a1 1 0 01.75-.97l6.98-2A1 1 0 0111 3v1z"
                ></path>
                <path
                  className="text-gray-400 fill-current"
                  d="M18.59 11l-1.3-1.3c-.94-.94.47-2.35 1.42-1.4l3 3a1 1 0 010 1.4l-3 3c-.95.95-2.36-.46-1.42-1.4l1.3-1.3H14a1 1 0 010-2h4.59z"
                ></path>
              </svg>
            </button>
          );
        }}
      </Mutation>
    );
  }
}
