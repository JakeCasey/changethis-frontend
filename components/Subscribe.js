import React from 'react';
import { Mutation } from 'react-apollo';
import StripeCheckout from 'react-stripe-checkout';
import { CURRENT_USER_QUERY } from './wrappers/User';
import { STRIPE_PUBLIC_KEY } from '../config';
import SickButton from './styles/SickButton';

import gql from 'graphql-tag';

const SUBSCRIBE_USER_MUTATION = gql`
  mutation SUBSCRIBE_USER_MUTATION($tokenId: String!, $planId: String!) {
    subscribe(tokenId: $tokenId, planId: $planId) {
      message
    }
  }
`;

export default class Subscribe extends React.Component {
  onToken = async (token, planId, subscribe) => {
    var { id } = token;
    const res = await subscribe({
      variables: {
        tokenId: id,
        planId: planId,
      },
    });
  };

  render() {
    let {
      plan: { planId, description },
    } = this.props;
    return (
      <Mutation
        mutation={SUBSCRIBE_USER_MUTATION}
        refetchQueries={[{ query: CURRENT_USER_QUERY }]}
      >
        {(subscribe, { loading, error }) => (
          <div>
            <h2>{planId}</h2>
            <StripeCheckout
              name={planId}
              description={description}
              token={token => this.onToken(token, planId, subscribe)}
              stripeKey={STRIPE_PUBLIC_KEY}
            >
              <SickButton disabled={loading}>
                Subscrib{loading ? 'ing' : 'e'}
              </SickButton>
            </StripeCheckout>
          </div>
        )}
      </Mutation>
    );
  }
}
