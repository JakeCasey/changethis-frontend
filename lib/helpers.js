const waitForIframeScrollPosition = (document, callback) => {
  let element = document
    .getElementById("iframe")
    .contentWindow.document.getElementById("iframeScrollPosition");

  if (typeof element === "undefined") {
    let interval = setInterval(() => {
      console.log("interval hit");
      //get element again
      try {
        element = document
          .getElementById("iframe")
          .contentWindow.document.getElementById("iframeScrollPosition");
      } catch (error) {
        if (error) console.log(error);
      }
      //check if it exists yet.
      if (typeof element !== "undefined") {
        //if it does kill the interval and run callback
        clearInterval(interval);
        console.log("Interval cleared");
        callback();
      }
      //if it doesn't keep checking.
    }, 500);
  } else {
    callback();
  }
};
export { waitForIframeScrollPosition };
