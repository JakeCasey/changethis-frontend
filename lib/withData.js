import withApollo from 'next-with-apollo';
import ApolloClient from 'apollo-boost';
import { endpoint, prodEndPoint } from '../config';
import { onError } from 'apollo-link-error';

const errorLink = onError(({ graphQLErrors, networkError }) => {
  if (networkError) {
    // Check if error response is JSON
    try {
      JSON.parse(networkError.bodyText);
    } catch (e) {
      // If not replace parsing error message with real one
      networkError.message = networkError.bodyText;
    }
  }
});

function createClient({ headers }) {
  return new ApolloClient({
    uri: process.env.NODE_ENV === 'development' ? endpoint : prodEndPoint,
    request: operation => {
      operation.setContext({
        fetchOptions: {
          credentials: 'include',
        },
        headers,
      });
    },
    //local data
    clientState: {
      resolvers: {},
      defaults: {
        cartOpen: false,
      },
    },
  });
}

export default withApollo(createClient);
