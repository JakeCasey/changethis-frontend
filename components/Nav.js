import Link from 'next/link';
import NavStyles from './styles/NavStyles';
import User from './wrappers/User';
import SignoutButton from './SignoutButton';
import styled from 'styled-components';

const Nav = () => (
  <User>
    {({ data, error, loading }) => {
      if (loading) return <p> loading ... </p>;
      let me = data.me;

      return (
        <NavStyles data-test="nav">
          {me && (
            <>
              <Link href="/permissions">
                <a>Permissions</a>
              </Link>
              <Link href="/plans">
                <a>Plans</a>
              </Link>
              <Link href="/reporting">
                <a>Reporting</a>
              </Link>
              <SignoutButton />
            </>
          )}
          {!me && (
            <>
              <Link href="/pricing">
                <a>Sign Up</a>
              </Link>
              <Link href="/signin">
                <a className="menu_button">Sign In</a>
              </Link>
            </>
          )}
        </NavStyles>
      );
    }}
  </User>
);

export default Nav;
