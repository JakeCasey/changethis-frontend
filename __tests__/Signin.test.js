import { mount } from 'enzyme';
import wait from 'waait';
import toJSON from 'enzyme-to-json';
import { MockedProvider } from 'react-apollo/test-utils';
import Signin, { SIGNIN_MUTATION } from '../components/Signin';
import { CURRENT_USER_QUERY } from '../components/User';
import { fakeUser } from '../lib/testUtils';
import { ApolloConsumer } from 'react-apollo';

function type(wrapper, name, value) {
  let input = wrapper
    .find(`input[name="${name}"]`)
    .simulate('change', { target: { name, value } });
}

const me = fakeUser();
const mocks = [
  //signup mock mutation
  {
    request: {
      query: SIGNIN_MUTATION,
      variables: {
        email: me.email,
        password: 'wes',
      },
    },
    result: {
      data: {
        signin: {
          __typename: 'User',
          id: 'abc123',
          email: me.email,
          name: me.name,
        },
      },
    },
  },
  //current user query mock
  {
    request: { query: CURRENT_USER_QUERY },
    result: {
      data: {
        me,
      },
    },
  },
];
describe('<Signin/>', () => {
  it('renders and matches snapshot', async () => {
    const wrapper = mount(
      <MockedProvider>
        <Signin />
      </MockedProvider>,
    );
    expect(toJSON(wrapper.find('form'))).toMatchSnapshot();
  });
  it('calls the mutation properly', async () => {
    let apolloClient;
    const wrapper = mount(
      <MockedProvider mocks={mocks}>
        <ApolloConsumer>
          {client => {
            apolloClient = client;
            return <Signin />;
          }}
        </ApolloConsumer>
      </MockedProvider>,
    );
    await wait();
    wrapper.update();

    type(wrapper, 'email', me.email);
    type(wrapper, 'password', 'wes');

    wrapper.update();
    wrapper.find('form').simulate('submit');
    await wait();
    //query the user out of the apollo client
    const user = await apolloClient.query({ query: CURRENT_USER_QUERY });
    expect(user.data.me).toMatchObject(me);
  });
});
