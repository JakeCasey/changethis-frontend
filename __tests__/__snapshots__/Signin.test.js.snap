// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Signin/> renders and matches snapshot 1`] = `
<form
  className="Form-sc-1oofk7n-0 gWcZGq"
  method="post"
  onSubmit={[Function]}
>
  <div
    className="img-form"
  >
    <div
      className="img-container"
    >
      <img
        src="/static/images/login.jpg"
      />
    </div>
    <div
      className="form-container"
    >
      <fieldset
        aria-busy={false}
        disabled={false}
      >
        <h2>
          Sign In
        </h2>
        <DisplayError
          error={Object {}}
        />
        <label
          htmlFor="email"
        >
          Email
          <input
            name="email"
            onChange={[Function]}
            placeholder="email"
            type="email"
            value=""
          />
        </label>
        <label
          htmlFor="password"
        >
          Password
          <input
            name="password"
            onChange={[Function]}
            placeholder="password"
            type="password"
            value=""
          />
        </label>
        <button
          type="submit"
        >
          Sign In
        </button>
      </fieldset>
    </div>
  </div>
</form>
`;
